### PR TITLE
First commit with agent

### PR DESCRIPTION
**Pull request by T-Kukawka**
This PR:

- adds/changes/removes etc

### Testing

Description on how oms-agent-app can be tested.

- [x] fresh install works
  - [x] Azure

#### Other testing

Description of features to additionally test for {APP-NAME} installations.

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Make sure `values.yaml` and `values.schema.json` are valid.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,17 +1,18 @@
 version: 2.1
 orbs:
-  architect: giantswarm/architect@2.11.0
+  architect: giantswarm/architect@3.3.0
```
Comment (ubergesundheit) [inline .circleci/config.yml:3]: This is fine, latest is 4.0.0 though (but no changes to the `push-to-app-catalog` job

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
   package-and-push-chart-on-tag:
     jobs:
       - architect/push-to-app-catalog:
           context: "architect"
           # executor: "app-build-suite" # uncomment this if you want automatic metadata generation
-          name: "package and push {APP-NAME} chart"
+          name: "package and push oms-agent chart"
           app_catalog: "giantswarm-playground-catalog"
           app_catalog_test: "giantswarm-playground-test-catalog"
-          chart: "{APP-NAME}"
```
Comment (ubergesundheit) [inline .circleci/config.yml:14]: duplicate line

File: helm/oms-agent/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+description: A Helm chart for oms-agent
+home: https://github.com/giantswarm/oms-agent-app
+icon: https://raw.githubusercontent.com/microsoft/Docker-Provider/ci_prod/img/azuremonitor-containers.svg
+name: oms-agent
+sources:
+  - https://github.com/microsoft/Docker-Provider/tree/ci_prod
+  - https://docs.microsoft.com/en-us/azure/monitoring/monitoring-container-health
+restrictions:
+  compatibleProviders: ["azure"]
+version: [[ .Version ]]
```
Comment (ubergesundheit) [inline helm/oms-agent/Chart.yaml:12]: Would be nice if you would keep this line aligned with the actual helm chart version

File: helm/oms-agent/values.yaml [hunk 4]
```diff
@@ -0,0 +1,133 @@
+# TEMPLATE-APP: This is set as a reasonable default, feel free to change.
+
```
Comment (ubergesundheit) [inline helm/oms-agent/values.yaml:2]:
```suggestion
```

File: helm/oms-agent/values.yaml [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+name: oms-agent
+namespace: oms-agent
+
+omsagent:
+  image:
+    repository: quay.io/giantswarm/ciprod
+    tag: ciprod06112021
```
Comment (ubergesundheit) [inline helm/oms-agent/values.yaml:9]: app name is `oms-agent` but container image name is `ciprod`?

**Review by ubergesundheit — CHANGES_REQUESTED** [review]
Looks okay, I think this is ready to be approved once all other comments are adressed.

Also please add an entry to the changelog.

File: helm/oms-agent/values.yaml (hunk 5 above)
Comment (T-Kukawka) [reply helm/oms-agent/values.yaml:9]: yes that is the MS naming crap that i wanted to have aligned at least unfortunately

File: helm/oms-agent/Chart.yaml (hunk 3 above)
Comment (T-Kukawka) [reply helm/oms-agent/Chart.yaml:12]: what do you mean?

Comment (ubergesundheit) [reply helm/oms-agent/Chart.yaml:12]: You're probably planning to release a 0.1.0 after merging this PR, the `[[``]]` templates are only used by `architect`, so not required here.
I suggest setting this to `0.1.0`
```suggestion
version: 0.1.0
```

File: helm/oms-agent/values.yaml (hunk 5 above)
Comment (ubergesundheit) [reply helm/oms-agent/values.yaml:7]: ok :+1:

File: .circleci/config.yml (hunk 2 above)
Comment (T-Kukawka) [reply .circleci/config.yml:14]: it is removed :D

File: .circleci/config.yml (hunk 1 above)
Comment (T-Kukawka) [reply .circleci/config.yml:3]: Done

File: helm/oms-agent/Chart.yaml (hunk 3 above)
Comment (T-Kukawka) [reply helm/oms-agent/Chart.yaml:12]: Done

**Comment by T-Kukawka** [comment]
@ubergesundheit I addressed all comments - please check changelog as am not sure what first entry there should be :D

File: README.md [hunk 6]
```diff
@@ -34,31 +52,68 @@ Here is an example that would install the app to
...
 Some apps have restrictions on how they can be deployed.
 Not following these limitations will most likely result in a broken deployment.
 
-*
+* Log Analytics Workspace ID and Key is a requirement.
 
 ## Credit
 
-* {APP HELM REPOSITORY}
+* https://github.com/microsoft/OMS-Agent-for-Linux
```
Comment (ubergesundheit) [inline README.md:119]: The chart is from here, right?

https://github.com/microsoft/Docker-Provider/tree/ci_prod/charts/azuremonitor-container

File: helm/oms-agent/Chart.yaml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+apiVersion: v1
+appVersion: 0.0.1
```
Comment (ubergesundheit) [inline helm/oms-agent/Chart.yaml:2]: Does the oms-agent have a version? `appVersion` should reflect this version.

File: README.md (hunk 6 above)
Comment (T-Kukawka) [reply README.md:119]: yes yes those are the 'releases' of MS

File: helm/oms-agent/Chart.yaml (hunk 7 above)
Comment (T-Kukawka) [reply helm/oms-agent/Chart.yaml:2]: yeah the version - ciprodsmthsmth

Comment (T-Kukawka) [reply helm/oms-agent/Chart.yaml]: Fixed

File: README.md (hunk 6 above)
Comment (T-Kukawka) [reply README.md]: fixed

**Pull request merged**